test/media/streaming_engine_unit.js

```diff
@@ -3041,6 +3041,33 @@ describe('StreamingEngine', () => {
       expect(secondRequest.abort).toHaveBeenCalled();
     });
 
+    it('aborts segment without size', async () => {
+      const videoStream = manifest.periods[0].variants[0].video;
```

Comment:
You should get `variants[1]` here instead of `variants[0]`. That will modify the variant we are trying to adapt into.